ui/component/core/src/index.ts

```diff
@@ -863,6 +966,16 @@ export class Manager implements EventProviderFactory {
         return this._basicIdentity ? this._basicIdentity.token : undefined;
     }
 
+    // Checking (manually) if token is invalid by verifying expiry date.
+    // By default, using a margin / minimum validity of 2 seconds
+    public isTokenExpired(margin = 2000): boolean {
```

Comment:
Just a point to note that server and client times may be different but not much you can do about that